File: 13.cpp
#include<iostream>
#include<fstream>
#include<vector>
using namespace std;



int main(){

ifstream file("13text.txt");

string buffer;
vector<int> sum(50);
for(int j=0;j<50;j++){sum[j]=0;}


while(!(file.eof())){
	
	getline(file,buffer);
	
	for(int i=0;i<50;i++){
		sum[i] += (buffer[i]-48);
	}
	
	//236
	//19360
	//4240
}

for(int i=49;i>=40;i--){
	sum[i-2] += (sum[i]/100);
	sum[i-1] += (sum[i]/10)%10;
	sum[i] = sum[i]%10;
}

for(int j=0;j<50;j++){cout<<sum[j]<<" ";}
	
}

File: 13.py
def carry(num):
    for i in range(len(num)-1,0,-1):
        if num[i]>9:
            num[i-1] += int(num[i]/10)
            num[i] = num[i]%10
    return num

def add(num,buffer):
    for i in range(50):
        num[i] += int(buffer[i])
    num = carry(num)
    return num

def solution():
    f = open('13text.txt','r')
    num = [0 for i in range(50)]
    for i in range(100):
        buffer = f.readline()
        buffer = buffer[0:-1]
        num = add(num,buffer)
    return num
for i in solution():
    print(i,end='')
#print(carry([1,2,3,40,90]))